src/main/java/Duke.java

```diff
@@ -1,10 +1,227 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
 public class Duke {
+    private static int taskCount;
+    private static ArrayList<Task> taskList = new ArrayList<>();
+    private final static String FILE_PATH = "data/duke.txt";
+    public static File f = new File(FILE_PATH);
+
     public static void main(String[] args) {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        System.out.println("\t____________________________________________________________");
+        System.out.println("\tHello! I'm Duke");
+        System.out.println("\tWhat can I do for you?");
+        System.out.println("\t____________________________________________________________");
+        //Load saved tasks from file
+        if (f.exists()) {
+            try {
+                loadProgress();
+            } catch (FileNotFoundException e) {
+                System.out.println("\tFile not found!");
+            }
+        } else {
+            try {
+                f.createNewFile();
+            } catch (IOException e) {
+                System.out.println("\tFile not created");
+            }
+        }
+
+        Duke.executeCommand();
+
+        //Save taskList to file
+        try {
+            writeToFile(FILE_PATH);
+        } catch (IOException e) {
+            System.out.println("\tError saving tasks!");
+        }
+    }
+
+    // Recognise command passed to Duke
+    public static void executeCommand() {
+        // Execute different response for different types of command
+        String line;
+        Scanner in = new Scanner(System.in);
+        int taskNumber;
+        while (true) {
+            line = in.nextLine();
+            if (line.equalsIgnoreCase("bye")) {
+                break;
+            }
+            if (line.equalsIgnoreCase("list")) {
+                displayList();
+                continue;
+            }
+            if (line.contains("done")) {
```

Comment:
You use contains() and a list of if statements to check the command. What if the command is "done delete"? The application might work find as the command is invalid. But it will also invoke the content of the next if statement. I think you should probably handle this by checking the command at its first a few characters.